python/lsst/multiprofit/plotting/plot_model_rgb.py

```diff
@@ -0,0 +1,398 @@
+
+    Parameters
+    ----------
+    model
+        The model to plot. If None, a dict of observations by band may be
+        passed as an additional kwarg; otherwise, only the data will be
+        plotted.
+    weights
+        Linear weights to multiply each band's image by.
+    high_sn_threshold
+        If non-None and given a model, this will return an image with the
+        pixels having a model S/N above this threshold in every band.
+    plot_singleband
+        Whether to make grayscale plots for each band.
+    plot_chi_hist
+        Whether to plot histograms of the chi (scaled residual) values.
+    chi_max
+        The maximum absolute value of chi in residual plots. Values of 3-5 are
+        suitable for good models while inadequate ones may need larger values.
+    rgb_min_auto
+        Whether to set the minimum in RGB plots automatically. Cannot supply
+        minimum in kwargs if enabled.
+    rgb_stretch_auto
+        Whether to set the stretch in RGB plots automatically. Cannot supply
+        stretch in kwargs if enabled.
+    **kwargs
+        Additional keyword arguments to pass to make_lupton_rgb when creating
+        RGB images.
+
+    Returns
+    -------
+    fig_rgb
+        The Figure for the RGB plots.
+    ax_rgb
+        The Axes for the RGB plots.
+    fig_gs
+        The Figure for the grayscale plots.
+    ax_gs
+        The Axes for the grayscale plots.
+    mask_inv_highsn
+        The inverse mask (1=selected) if high_sn_threshold was specified.
+    """
+    if rgb_min_auto and "minimum" in kwargs:
+        raise ValueError(f"Cannot set rgb_min_auto and pass {kwargs['minimum']=}")
+    if rgb_stretch_auto and "stretch" in kwargs:
+        raise ValueError(f"Cannot set rgb_stretch_auto and pass {kwargs['stretch']=}")
+    if not (chi_max > 0):
+        raise ValueError(f"{chi_max=} not >0")
+    if weights is None:
+        if model is None:
+            weights = {band: 1.0 for band in kwargs["observations"].keys()}
+        else:
+            bands_set = set()
+            bands = []
+            weights = {}
+            for obs in model.data:
+                band = obs.channel.name
+                if band not in bands_set:
+                    bands_set.add(band)
+                    bands.append(band)
+                    weights[band] = 1.0
+
+    n_data = len(model.data)
+    has_model = model is not None
+    observations = {}
+    models = {}
+
+    if has_model and (n_data < 3):
+        if n_data == 1:
+            # pretend this is three bands
+            obs, output_data = model.data[0], model.outputs[0].data
+            band = obs.channel.name
+            weights = {}
+            for idx in range(1, 4):
+                key = f"{band}{idx}"
+                weights[key] = 1.0
+                observations[key] = obs
+                models[key] = output_data
+        elif n_data == 2:
+            raise NotImplementedError("RGB images for two-band data are not supported (yet)")
```

Comment:
See https://github.com/lsst/scarlet_lite/blob/main/python/lsst/scarlet/lite/display.py#L41-L116 (credit to Peter Melchior) for one possible solution to mapping between 1 and 7 channels into RGB.

Reply:
Yeah, I saw that and the code further down ends up doing the same thing. I don't remember why I did it in a way that doesn't work for two bands but I'll fix it when someone (possibly future me) complains enough.